web/components/SchedulePage.jsx

```diff
@@ -1,33 +1,149 @@
 import React from 'react';
+import moment from 'moment';
+import { connect } from 'react-redux';
 
-export default class SchedulePage extends React.Component {
-    renderBaseOnParams = (match) => {
-        if (match.params.teamMemberId) {
+import ScheduleHeader from './ScheduleHeader';
+import SideBar from './SideBar';
+import Flex from './Flex';
+import SetBlock from './SetBlock';
+import Text from './Text';
+
+import { fetchCurrentTeamMemberById, setSelectedDay } from '../reducers/environment';
+import LoadingDots from './Loading';
+
+class SchedulePage extends React.Component {
+    state = {
+        daysOfWeek: [],
+    }
+    
+    componentDidMount() {
+        const { history, match } = this.props;
+        const teamMemberId = match.params.teamMemberId || 'recGVSamjigJbZoJ8';// mocked id until to have a login
+        this.props.fetchCurrentTeamMemberById({ id: teamMemberId })
+        this.getDaysOfWeek()
+        if (!match.params.teamMemberId) {
+            const today = moment().toDate();
+            // If the match.params don't have a teamMemberId are u seeing your schedule
+            history.push('/schedule/' + today.getDay());
+            // /schedule - SchedulePage, have today's day selected by default
+            this.props.setSelectedDay(today)
+        }
+    }
+
+    renderSetBlocks = (selectedDay) => {
+        const { currentTeamMember } = this.props;
+        selectedDay = moment(selectedDay).format('YYYY-MM-DD')
+        const setBlocksByDate = _.groupBy(currentTeamMember.weeklySetblocks, 'date')
+        const setBlocks = setBlocksByDate[selectedDay];
+        if (setBlocks) {
+            return setBlocks.map(setBlock => {
+                return <SetBlock data={setBlock} key={setBlock.id} />
+            })
+        } else {
             return (
-                <h6>
-                teamMemberId:
-                    {match.params.teamMemberId}
-                </h6>
+                <Text weight='600' aling='center'> This user hasn't committed any Setblocks for this day </Text>
             )
         }
+    }
+
+    getDaysOfWeek = () => {
+        const startOfWeek = moment()
+        .startOf('isoWeek');
+        const endOfWeek = moment()
+        .endOf('isoWeek');
+
+        const days = [];
+        let day = startOfWeek;
+
+        while (day <= endOfWeek) {
+            days.push(day.toDate());
+            day = day.clone()
+            .add(1, 'd');
+        }
+
+        this.setState({
+            daysOfWeek: days,
+        })
+        this.props.setSelectedDay(days[0])
+    }
 
-        if (match.params.dayOfWeek) {
+    renderIfItReady() {
```

Comment:
I think there's a cleaner way to implement loaders I'll have to dig it up from an old repo, but we can revisit this when we are closer to a stable release it's fine as is 👍